docs/DeveloperGuide.md

```diff
@@ -1,11 +1,308 @@
+        1. [Delete Task](#delete-task)
+        1. [Change Task Priority](#change-task-priority)
+        1. [Mark Task as Complete](#mark-task-as-complete)
+    1. [Sprint](#sprint)
+        1. [Create Sprint](#create-sprint)
+        1. [View Sprint](#view-sprint)
+        1. [Add Task to Sprint](#add-task-to-sprint)
+        1. [Remove Task from Sprint](#remove-task-from-sprint)
+        1. [Allocate Sprint Tasks to Members](#allocate-sprint-tasks-to-members)        
+    1. [Storage](#storage)
+        1. [Location](#location)
+        1. [Loading Data](#loading-data)
+            1. [Converting and Mapping of JSON to Objects](#converting-and-mapping-of-json-to-objects)
+        1. [Saving Data](#saving-data)
+            1. [When the Program Exits](#when-the-program-exits)
+            1. [Changes Made to the Data](#changes-made-to-the-data)
+            1. [Serialising Objects to JSON](#serialising-objects-to-json)
+1. [Appendix: Requirements]()
+1. [Others](#target-user-profile)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Introduction
+### Background
+SCRUMptious is a Java-based command line interface application for you to efficiently manage the development of a project. Leveraging the robust SCRUM/Agile framework, it allows you to delegate tasks to your team members and organize project requirements with ease. As a bonus, if you are a keyboard warrior, you can reach peak efficiency using SCRUMptious to manage your projects.
 
+### Purpose
+This guide illustrates the general architecture, and software design of SCRUMptious.
+
+### Scope
+This guide is geared towards developers who wish to enhance or create their own version of SCRUMptious. As such, it contains important information regarding the software architecture and design considerations of SCRUMptious.
+
+## Getting Started
+### Prerequisites
+1. JDK 11.
+1. IntelliJ IDEA.
+
+### Setting Up
+1. Use a Git tool to fork this repository, or download the .zip file from GitHub and extract the contents into a new folder.
+1. Right-Click on the folder and select “Open folder as Intellij IDEA Community Edition Project”.
+1. Ensure JDK 11 is selected for Gradle, by navigating to Configure > Structure for New Projects > Project Settings > Project > Project SDK.
+
+### Running the Program
+This program can be run once it is compiled. If you have built its artifacts (.jar) file, you may run it using java -jar <filename.jar> on your command line.
+
+## Design
+This section seeks to explain the high-level design of the application. Given below is a quick overview of each component and the explanation of the design architecture in greater detail.
+NotUS is the main class of the application, and handles the initializing and execution of the appropriate classes.
+### Architecture
+![Figure X: Architecture Diagram](./image/developerguide/architecturediagram.png "Architecture Diagram UML")  
```

Comment:
I like how simple and easy your architecture diagram looks, but I am a bit lost when the user points to UI and there's no arrow pointing out of the UI, it feel like the flow wasn't complete. I thought that it could be better if the UI points to the user, it would make the flow continuous as well.

Reply:
To clarify, there should be two arrows one from the user to the UI and one from UI to the user?